### DISCUSSION TITLE
how to exclude everything except

### DISCUSSION THREAD

**Discussion by hesxenon** [borgbase/vorta#2089] [FAQ] [answered]
I'm trying to exclude every hidden folder in home dirs (like those created by e.g. java, rust, deno, many other programs etc) _except_ .ssh

The reason is, that the hidden files I am actually interested in are tracked in dotfiles but I don't want to push my private keys to some repo.

In general it would be nice to have some sort of "textual config editing" - just providing a link to borgbase docs when you can't actually see what patterns and exclusions are being used doesn't seem to be enough to me.

**Comment by goebbe** [comment]
You can use _excludes_ and _includes_ by using `--pattern` or `--patterns-from`  options, via Vorta > Schedule > Shell Commands > Extra arguments for borg create.

Feature request with examples for both options: https://github.com/borgbase/vorta/issues/1991

The following "Extra arguments for borg" will include .ssh but exclude all other hidden files and folders in /home/username (and subfolders): 
`--pattern=+ home/username/**/.ssh   --pattern=- home/username/**/.*`
Order matters! First specify the include "+" pattern, then specify the exclude "-" pattern. 

Now, rereading your comment: It may be that you got confused about the meaning of "includes" and "excludes". 
If the only thing you want to achieve is to **exclude** .ssh from your backup, but include all other hidden files and folders, you could simply go to Vorta > Source > Manage Excluded items
Push the "+" button (here, this is to add a new exclude pattern) and type `.ssh`   
All other hidden files and folders should be included in the backups, anyway, if you use `/home/username` as a source.  

Hope this helps.

**Comment by hesxenon** [comment]
thx, but I don't really understand what I would specify in "sources" then? Nothing? Root? 

Is Vorta even a good choice for this use case?

**Reply by goebbe** [reply]
For the use case: Keep in mind that Vorta aims to be a tool for the backup of _user data._
If you are concerned about backup of _system data_ you should consider using borg directly.

**Comment by goebbe** [comment]
@hesxenon  
In sources you specify the path(s) of the folders you want to put into your backup. 
e.g. `/home/hexcenon`
By default, Vorta will do a backup of all the files and subfolders in this path(s). This includes all hidden files/ folders. 
Now, if you want to exclude specific files/ folder from your backup you can use "exclude" rules.  
Vorta > Source > Manage Excluded items (at the bottom of this tab)

So you have (at least) two different ways to choose your folders/ files for the backup: 
1. Select all folders paths manually and add them in sources
or: 
2. Select a "parent" path e.g. your home directory in sources and exclude the files/ folders that you do not want to have in your backup (by using exclude patterns). 

You can mix the two approaches. :-)

**Reply by hesxenon** [reply]
yeah, but with the "sources/exclude" approach I cannot specifiy the additional create args, can I? And thus I don't know how to exclude .ssh while specifying exclude items

**Reply by goebbe** [reply, marked as answer]
If you want to use a list of includes and excludes, I recommend using the `--patterns-in` option and point to a text-file where all the sources / includes and excludes can easily be specified in a long list. 
Have a look at the following post in this feature request, for an example: https://github.com/borgbase/vorta/issues/1991#issuecomment-2354851384

You can _combine_ the --patterns-in option with additional create options. As far as I know, there are no limitations, here. 

These patterns-files are referred to at the level of Vorta profiles and are applied each time you create a new backup. 
You should also have a look at the borg specs for more details: https://borgbackup.readthedocs.io/en/stable/usage/help.html   But keep in mind that borg supports different pattern styles, which can be confusing.
